kasa/iot/modules/module.py

```diff
@@ -24,15 +26,18 @@ def merge(d, u):
     return d
 
 
-class Module(ABC):
+class Module(BaseModule, ABC):
     """Base class implemention for all modules.
 
     The base classes should implement `query` to return the query they want to be
     executed during the regular update cycle.
     """
 
-    def __init__(self, device: "SmartDevice", module: str):
-        self._device: "SmartDevice" = device
+    def __init__(self, device: "Device", module: str):
+        if TYPE_CHECKING:
+            self._device: IotDevice = cast(IotDevice, self._device)
+        else:
+            self._device = device
```

Comment:
I don't think one should use else clauses with TYPE_CHECKING..

I'm wondering if we shouldn't also move the `Module` to become a `BaseModule` already in this PR?

Reply:
Removed the type checking.

Maybe we should hold off renaming `Module` > `BaseModule` until we're discussing the module interface and we can then decide whether `kasa.Module` > `kasa.BaseModule` as well as `kasa.iot.Module` > `kasa.iot.BaseModule` etc.

Reply:
Else clause is removed and `Module` renamed to `BaseModule`